flow/src/org/labkey/flow/reports/statPicker.jsp

```diff
@@ -93,16 +96,9 @@
     }
     stats.append("}");
 
-    StringBuilder ops = new StringBuilder();
-    ops.append("[");
-    comma = "";
-    for (CompareType ct : new CompareType[] { CompareType.EQUAL, CompareType.NEQ_OR_NULL, CompareType.ISBLANK, CompareType.NONBLANK, CompareType.GT, CompareType.LT, CompareType.GTE, CompareType.LTE, CompareType.CONTAINS, CompareType.STARTS_WITH, CompareType.DOES_NOT_CONTAIN, CompareType.DOES_NOT_START_WITH, CompareType.IN })
-    {
-        ops.append(comma);
-        ops.append("[\"").append(ct.getPreferredUrlKey()).append("\", \"").append(ct.getDisplayValue()).append("\"]");
-        comma = ",\n";
-    }
-    ops.append("]");
+    var compareTypesToDisplay = Stream.of(CompareType.EQUAL, CompareType.NEQ_OR_NULL, CompareType.ISBLANK, CompareType.NONBLANK, CompareType.GT, CompareType.LT, CompareType.GTE, CompareType.LTE, CompareType.CONTAINS, CompareType.STARTS_WITH, CompareType.DOES_NOT_CONTAIN, CompareType.DOES_NOT_START_WITH, CompareType.IN )
+            .map(ct -> new String[] {ct.getPreferredUrlKey(), ct.getDisplayValue()});
+    JSONArray ops = toJsonArray(compareTypesToDisplay.collect(Collectors.toList()));
```

Comment:
Why not use toJsonArray(Stream) here?

(Even cooler would be to implement a JSONArrayCollector. See Collectors.of(). I'm not really serious... Collectors are a bit mind-bending to implement. See MultiValuedMapCollectors.java for an example.)

Reply:
Sign... I "accidentally" implemented JSONArray.collector(). I'll commit it to this branch...

Reply:
Haha nice! I also tried to use JsonArray(Stream) but it didn't work too well since the object passed in is `Stream<String[]>` and it's looking for a `Stream<Object>`. I cast the array as an object though (which looked a little funky), but it worked.